src/Model/Package.php

```diff
@@ -34,7 +34,35 @@ public function getTitle()
      */
     public function getSummary()
     {
-        return $this->renderWith('Package_summary');
+        $summary = $this->renderWith('Package_summary');
+        $this->extend('updateSummary', $summary);
+        return $summary;
+    }
+
+    /**
+     * Gives the summary template {@see getSummary()} a list of badges to show against a package
+     *
+     * badgeDefinitions are in the format [$title => $type] where:
+     *   title is the unique string to display
+     *   type is an optional class attribute (applied as a BEM modified, by default)
+     *
+     * @param array $badgeDefinitions
+     *
+     * @return ArrayList
+     */
+    public function getBadges($badgeDefinitions = [])
+    {
+        $this->extend('updateBadges', $badgeDefinitions);
```

Comment:
See [previous comment](https://github.com/bringyourownideas/silverstripe-maintenance/pull/72#discussion_r191298528) - I think this should go after the foreach.